model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,182 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Corresponds to a single survey.
+message Survey{
+  // The ID of the survey.
+  string survey_id = 1;
+
+  // A list of questions that make up the survey.
+  repeated SurveyQuestion questions = 2;
+}
+
+// Represents a question that is part of the Survey.
+message SurveyQuestion{
+  // The ID of the survey question.
+  string question_id = 1;
+
+  // The type of question.
+  QuestionType question_type = 2;
+
+  // The actual text value of the question.
+  string  question = 3;
+
+  // The semantic name of the question used to identify it in the analytics dashboard.
+  QuestionName question_name = 4;
+
+  // A list of multi-choice answer options for a given question.
+  repeated  SurveyQuestionAnswer answer_list = 5;
+}
+
+// The list of types a question can have that is used to determine how it is presented by the UI.
+enum QuestionType{
+  // The type of question is unknown.
+  QUESTION_TYPE_UNSPECIFIED=0;
+  // Corresponds to a question with multiple choices.
+  MULTIPLE_CHOICE = 1;
+
+  // Corresponds to a question with multiple choices and a free-text 'other' option.
+  MULTIPLE_CHOICE_WITH_OTHER = 2;
+
+  // Corresponds to a question which takes a free-text response.
+  FREE_FORM = 3;
+}
+
+// Corresponds to the semantic name given to each question.
+enum QuestionName{
+  // The question name is unknown.
+   QUESTION_NAME_UNSPECIFIED = 0;
+  // Corresponds to the name of the user type question.
+  QUESTION_USER_TYPE = 1;
+
+  //  Corresponds to the market fit question.
+  QUESTION_MARKET_FIT = 2;
+
+  // Corresponds to  the NPS question.
+  QUESTION_NPS = 3;
+
+  // Corresponds to  the open-ended question for a NPS promoter.
+  QUESTION_PROMOTER_FEEDBACK = 4;
+
+  // Corresponds to  the open-ended question for a NPS passive.
+  QUESTION_PASSIVE_FEEDBACK = 5;
+
+  // Corresponds to  the open-ended question for a NPS detractor.
+  QUESTION_DETRACTOR_FEEDBACK = 6;
+}
+
+// Represents the multiple choice answer available for a question.
+message SurveyQuestionAnswer{
+  // The ID of this answer option.
+  string question_answer_id = 1;
+
+  // The semantic name of the question this answer option is tied to.
+  QuestionName question_name = 2;
+
+  // The actual text value of the answer.
+  string answer = 3;
+}
+
+// Represents an answer that was selected by a user for a survey question.
+message SurveyUserAnswer{
+  //  The ID of the survey to which this answer is tied.
+  string  survey_id = 1;
+
+  // Semantic name of the question the response is tied to.
+  QuestionName question_name = 2;
+
+  // The value of the answer the user gave.
+  string user_answer = 3;
```

Comment:
Yes, it was meant to be the actual label text.
I see your point about translated text, and I think it might be adviasable to use the associated enums instead.

Reply:
Enums sound good if we always have a fixed set of text choices to choose from.

Reply:
At the moment, all the MCQs have a fixed set of answers, so I have opted to create a oneof of each of those answer representations that this field could take.

Reply:
Re: translations, I have added a language code field to the question so that we can query open-ended responses based on this. It occurred to me that a generic query will just show all the answers without the language context.

Reply:
Given the change, is this field still needed? I'm assuming the question type provides enough context to allow the UI to pick app strings to show for each question, so it isn't clear to me why this string will be needed.

Reply:
It is no-longer needed. Please counter-check, I removed it in a previous commit.